gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -207,6 +209,145 @@ def __repr__(self):
+        for VCF files is CompressionTypes.UNCOMPRESSED. If set to
+        CompressionTypes.AUTO, file_path's extension will be used to detect
+        compression.
+      headers: A list of VCF meta-information lines describing the INFO,
+        FILTER, and FORMAT entries in each record and a header line describing
+        the column names. These lines will be written at the beginning of the
+        file.
+    """
+    super(_VcfSink, self).__init__(
+        file_path,
+        coder,
+        num_shards=num_shards,
+        mime_type='text/plain',
+        compression_type=compression_type)
+    self._headers = headers
+
+  def open(self, temp_path):
+    file_handle = super(_VcfSink, self).open(temp_path)
+    if self._headers is not None:
+      for header in self._headers:
+        file_handle.write(header)
+    return file_handle
+
+  def _get_variant_info(self, variant):
+    info_list = []
+    # Set end in info if this is a non-variant.
+    if (variant.start and variant.reference_bases and
+        variant.start + len(variant.reference_bases) < variant.end):
+      info_list.append('END=%d' % variant.end)
+    # Set all other fields of info.
+    for k, v in variant.info.iteritems():
+      if v.field_count == '1':
+        info_list.append('%s=%s' % (k, str(v.data)))
+      else:
+        info_list.append('%s=%s' % (k, ','.join([str(x) for x in v.data])))
+    return ';'.join(info_list)
+
+  def _get_variant_format(self, variant):
+    format_keys = ['GT']
+    if variant.calls:
+      # If any calls have a set phaseset that is not `DEFAULT_PHASESET_VALUE`,
+      # the key will be added to the format field.
+      for call in variant.calls:
+        if call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE:
+          format_keys.append('PS')
+          break
+      format_keys.extend([k for k in sorted(variant.calls[0].info)])
+    return ':'.join(format_keys)
+
+  def _get_variant_calls(self, variant):
+    calls_values = []
+    alternate_phaseset = False
+
+    for call in variant.calls:
+      if call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE:
+        alternate_phaseset = True
+        break
+
+    for call in variant.calls:
+      # Set GT field.
+      genotype_field = MISSING_FIELD_VALUE
+      if call.genotype != [MISSING_GENOTYPE_VALUE]:
+        phase_char = '/' if call.phaseset is None else '|'
+        genotype_field = phase_char.join([str(x) for x in call.genotype])
+      call_info_fields = [genotype_field]
+      # Set PS field if there is a set phaseset that is not
+      # `DEFAULT_PHASET_VALUE`.
+      if alternate_phaseset:
+        if call.phaseset and call.phaseset != DEFAULT_PHASESET_VALUE:
+          call_info_fields.append(call.phaseset)
+        else:
+          call_info_fields.append(MISSING_FIELD_VALUE)
+      # Set all other fields of the call info.
+      for k in sorted(call.info):
+        if call.info[k]:
```

Comment:
We should check whether call.info[k] is None or empty instead (the value can be zero). Since we need to do the same below, how about creating an isNoneOrEmpty method? There may already be such a thing in some python library.

Reply:
Done.